rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapSupportImplTest.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsEqualToMaxLockingCap_shouldReturnTrue() {
+        // Arrange
+        Coin previousLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(previousLockingCap);
+        lockingCapSupport.save();
+        // Recreate LockingCapSupport to load the previous value from storage
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+
+        // The new locking cap is the maximum value that can be set
+        Coin newLockingCap = previousLockingCap.multiply(constants.getIncrementsMultiplier());
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertTrue(actualResult);
+        assertEquals(Optional.of(newLockingCap), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsEqualToLockingCapValueAlreadyExists_shouldReturnTrue() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue();
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertTrue(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsZero_shouldReturnFalse() {
+        // Arrange
+        Coin newLockingCap = Coin.ZERO;
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsNegative_shouldReturnFalse() {
+        // Arrange
+        Coin newLockingCap = Coin.NEGATIVE_SATOSHI;
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_prePapyrus200_whenLockingCapIsNotSet_shouldReturnFalse() {
```

Comment:
not sure about the `prePapyrus` thing. The implementation of the method returns false when the value got from `getLockingCap()` is empty. 

we knowing that this happens when papyrus is not active is another matter, related to `getLockingCap` implementation itself.

Reply:
It's not only a matter of the `getLockingCap` method. We test the following `IF` behavior in the `increaseLockingCap` method that happens in a `prePapyrus200` scenario:

`if (!currentLockingCap.isPresent())`

Reply:
that's exactly what i mean. You should test the behavior when currentLockingCap is not present. The fact that you know that this only happens pre papyrus is another matter, but its not known by reading `increaseLockingCap` method.

Reply:
I understand what you say. However, Marcos told me it's not a good idea to use mocks if not needed. That is our last choice. If we can recreate a test closely to real life we have to take it of that way.